### PR TITLE
Add ghost server logic

### PR DESCRIPTION
**Pull request by barnabwhy**
Allows servers to reregister with the same id without any change to launcher code

### CODE REVIEW

**Comment by barnabwhy** [comment]
Finally, functioning pr :trollface:

File: server/gameserver.js [hunk 1]
```diff
@@ -88,6 +88,82 @@ async function SharedTryAddServer( request )
...
+	let playerCount = request.query.playerCount || 0
+	if ( typeof playerCount == "string" )
+		playerCount = parseInt( playerCount )
+
+	if ( typeof request.query.maxPlayers == "string" )
+		request.query.maxPlayers = parseInt( request.query.maxPlayers )
+
+	let name = filter.clean( request.query.name )
+	let description = request.query.description == "" ? "" : filter.clean( request.query.description )
+	let newServer = new GameServer( name, description, playerCount, request.query.maxPlayers, request.query.map, request.query.playlist, request.ip, ghost.port, ghost.authPort, request.query.password, modInfo )
```
Comment (abarichello) [inline server/gameserver.js:153]: You should refactor this section into a common function since this is basically a copypaste from `SharedTryAddServer`

**Review by abarichello — CHANGES_REQUESTED** [review]

**Comment by barnabwhy** [comment]
image
fuck

**Review by abarichello — APPROVED** [review]
Thanks

**Pull request merged**